### ISSUE TITLE
performance horrible in comparison with scratch itself

### ISSUE THREAD

**Issue by arch-user-france1** [SheepTester/htmlifier#146]
The performance is horrible. Take a look at miner cat 4. It wont run on your computer with this.

**Comment by ShipmasterKyle** [comment]
I think your computer is just bad. HTMLifier converts scratch projects to an html file. Large games tend to take longer to load or crash on low end computers. Thats not really an HTMLifier issue.

**Comment by debian-user-france1** [comment]
My computer is not bad at all it's able to handle ark at max settings with 60 FPS.

I said it's extremely bad and I compared with the real scratch. It almost crashed the browser.

**Comment by ShipmasterKyle** [comment]
Thats not htmlifier performance though. Your issue is your **browser** crashed not HTMLifier crashed. Seems like your browser can't handle the HTML file. What settings did you use to HTMLify?

**Comment by SheepTester** [comment]
The HTML file requires more memory because all the assets are stored as strings in the HTML which increases their file size by a third as well as stores them in memory. Does using the [download as a zip](https://github.com/SheepTester/htmlifier/wiki/Downloading-as-a-.zip) option help?

**Comment by Ganheso** [comment]
> The HTML file requires more memory because all the assets are stored as strings in the HTML which increases their file size by a third as well as stores them in memory. Does using the [download as a zip](https://github.com/SheepTester/htmlifier/wiki/Downloading-as-a-.zip) option help?
> 
> This is a consequence of people's naive desire to want a single HTML file

did anyone actually ask for just a single file? I prefer downloading as a zip simply cuz it looks "cooler"

**Comment by SheepTester** [comment]
A lot of Scratchers think in terms of "converting" rather than "packaging," and they expect the file extension to change to .html. Also, the HTML file in the zip file can't be directly opened in the browser

**Comment by Ganheso** [comment]
> the HTML file in the zip file can't be directly opened in the browser
I think I just found out why my old projects dont work anymore

**Comment by ShipmasterKyle** [comment]
If you want to play the game as a zip you could always simply unpack it in a folder then run the index.html included.

**Comment by debian-user-france1** [comment]
> Thats not htmlifier performance though. Your issue is your **browser** crashed not HTMLifier crashed. Seems like your browser can't handle the HTML file. What settings did you use to HTMLify?

Idk but my  browser is able to handle the script on scratch. I've just set FPS to 60 and maybe enabled Turbo Mode.

My Browser is chromium based.
Its unplayable with this.

**Comment by Ganheso** [comment]
> If you want to play the game as a zip you could always simply unpack it in a folder then run the index.html included.

no you can't, as Sean just said
I know this myself cus I tried doing it before, it just doesn't work

**Comment by debian-user-france1** [comment]
> The HTML file requires more memory because all the assets are stored as strings in the HTML which increases their file size by a third as well as stores them in memory. Does using the [download as a zip](https://github.com/SheepTester/htmlifier/wiki/Downloading-as-a-.zip) option help?
> 
> This is a consequence of people's naive desire to want a single HTML file

I used it with ZIP and a Python webserver

**Comment by debian-user-france1** [comment]
> If you want to play the game as a zip you could always simply unpack it in a folder then run the index.html included.

Most browser don't allow file-CORS requests so that won't work.

**Comment by ShipmasterKyle** [comment]
> > If you want to play the game as a zip you could always simply unpack it in a folder then run the index.html included.
> 
> no you can't, as Sean just said I know this myself cus I tried doing it before, it just doesn't work

Yeah I just tested this. I infered it should work as the html file is still executable with a file:// link. A further examination shows that it crashes on launch however I'm 99% sure ik a work around for this although I have yet to test it,
